code/modules/projectiles/guns/energy/dueling.dm

```diff
@@ -147,6 +164,23 @@
 	setting_overlay = mutable_appearance(icon,setting_iconstate())
 	add_overlay(setting_overlay)
 
+/obj/item/gun/energy/dueling/attackby(obj/item/W, mob/user, params)
+	if(istype(W, /obj/item/gun/energy/dueling))
+		var/obj/item/gun/energy/dueling/other_gun = W
+
+		if(!check_valid_duel(user, FALSE) && !other_gun.check_valid_duel(user, FALSE))
+			var/datum/duel/D = new(src, other_gun)
+			to_chat(user,span_notice("Pairing established. Pairing code: [D.pairing_code]"))
+
```

Comment:
return

Reply:
idgi

Reply:
you need to return after doing the thing, otherwise it's going to go through the rest of the attackby chain.